pkg/controller/configurationpolicy/configurationpolicy_controller.go

```diff
@@ -1230,34 +1230,16 @@ func mergeSpecsHelper(templateVal, existingVal interface{}, ctype string) interf
 			//if one field is a list and the other isn't, don't bother merging
 			return templateVal
 		}
-		if len(existingVal) > len(templateVal) {
-			//if there are more values in the existing object than the template and our complianceType is musthave,
-			//we need to merge in the extra data in the existing object to do a proper compare
-			if ctype != "mustonlyhave" {
-				return mergeArrays(templateVal, existingVal, ctype)
-			}
-			return templateVal
-		}
 		//if existing value is shorter than the template value, no merge needed since it will always be a mismatch
 		if len(existingVal) < len(templateVal) {
 			return templateVal
 		}
```

Comment:
I'm suspicious of this section now. Can we add a test that creates an oauth with 2 idps, then a new config policy that specifies 3 additional idps, and finally check how many the resulting oauth has? I suspect it will only be 3 - the old idps will be overwritten based on this condition, but don't we want all 5?

There could be something else ensuring it appends, so I'm not positive about this. Also, if you'd rather add that in a separate PR we can do that.